read-cache.c

```diff
@@ -2935,9 +2937,14 @@ static int do_write_index(struct index_state *istate, struct tempfile *tempfile,
 	if (!strip_extensions && !drop_cache_tree && istate->cache_tree) {
 		struct strbuf sb = STRBUF_INIT;
 
+		trace2_region_enter("index_ext", "cache_tree/write", NULL);
+
 		cache_tree_write(&sb, istate->cache_tree);
 		err = write_index_ext_header(&c, &eoie_c, newfd, CACHE_EXT_TREE, sb.len) < 0
```

Comment:
Maybe have a checkpoint in between, just in case writing the extension header has some funny behavior...

Reply:
good idea.  i've broken this into time to format the extension buffer and time to actually write it to disk.

Reply:
done. thanks!